src/docs/arch-hs.dox

```diff
@@ -1,8 +1,125 @@
 /*! @arch_page arch-hs History Store
 
-The History Store in WiredTiger tracks old (all but the latest committed)
-versions of records.  By having these records in storage separate from
-the current version, they can be used to service long running transactions,
+The history store in WiredTiger tracks historical versions of records required to
+service older readers. By having these records in storage separate from
+the current version, they can be used to service long running transactions
 and be evicted as necessary, without interfering with activity that uses
-the most recent committed versions.
+the most recent committed versions. With the introduction of history store, only the
+newest update to a key is written to the user table while the older updates
+for the key are written to the history store. All user tables in the database are
+backed by a single history store table. The history store has no outward
+visibility to the application code and only WiredTiger's internal modules can
+perform operations on the history store table using a predefined cursor API.
+
+@section arch_hs_table History store table structure
+WiredTiger writes the history store on the disk as a standard row-store table. The key for
+the history store table is a combination of:
+- \c btree ID of user table this update belongs to
+- \c record key (byte-string for row-store, record number for column-store)
+- \c start timestamp for the update
+- \c counter value
+This key format allows us to search efficiently for a given record key and read
+timestamp combination. The last part of the key is a monotonically increasing counter
+to keep the key unique in scenarios where a key has multiple updates with the same commit timestamp.
+As the key for the history store table is different for row- and column-store,
+we store both key types in a byte string otherwise we'd need two history store files with
+different key formats.
+
+The corresponding value for each key is stored as a combination of:
+- \c stop timestamp
+- \c durable timestamp
+- \c update type
+- \c value.
+
+@subsection arch_hs_table_tombstone Tombstones in the history store table
```

Comment:
Still there is no mention of garbage collection. I still think we need to mention the purpose of the stop timestamps is for garbage collection. No need to mention the details of garbage collection of course.

Reply:
I have discussed process of deleting updates/pages with globally visible tombstones in section for reconciliation. Do you think it will be better to mention them here as well?

Reply:
Or I can add something here like "see section reconciliation how we manage updates once they have become globally visible"?

Reply:
Yes, the second option is fine. But how is reconciliation related to garbage collection? Garbage collection is a separate thing running in checkpoint.

Reply:
Garbage collection is already talked about in the checkpoint architecture page so if @quchenhao you do want a section on that, perhaps haseeb can just reference it?

Reply:
Then reference is OK.

Reply:
I have added a reference to garbage collection in checkpoint. Check out 5d8232c